smart_card_connector_app/src/window-apps-displaying.js

```diff
@@ -47,21 +52,45 @@ var logger = GSC.Logging.getScopedLogger('ConnectorApp.MainWindow');
 var appListElement = goog.dom.getElement('app-list');
 
 /**
+ * @const
+ */
+var knownAppsRegistry = new PermissionsChecking.KnownAppsRegistry();
+
+/**
+ * @type {goog.Promise.<!Array.<PermissionsChecking.KnownApp>>}
+ */
+var lastKnownAppsPromise = null;
+
+/**
  * @param {!Array.<string>} appList
  */
 function onUpdateListener(appList) {
   logger.fine('Application list updated, refreshing the view. ' +
               'New list of id\'s: ' + appList);
 
-  GSC.Logging.checkWithLogger(logger, !goog.isNull(appListElement));
-  goog.asserts.assert(appListElement);
+  var knownAppsPromise = knownAppsRegistry.tryGetByIds(appList);
+
+  knownAppsPromise.then(
+    function(knownApps) {
+      if (knownAppsPromise !== lastKnownAppsPromise) return;
+
+      GSC.Logging.checkWithLogger(logger, !goog.isNull(appListElement));
+      goog.asserts.assert(appListElement);
+
+      goog.dom.removeChildren(appListElement);
 
-  goog.dom.removeChildren(appListElement);
+      for (var i = 0; i < knownApps.length; i++) {
+        var text = knownApps[i] ? knownApps[i].name : appList[i];
```

Comment:
Please add surrounding the extension id to distinguish this from the "normal" case when the name is displayed.
E.g.:
`<somelongextensionsid>`